### PR TITLE
remove unneeded chain function usages

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: src/ModalHeader.js [hunk 1]
```diff
@@ -51,21 +55,17 @@ class ModalHeader extends React.Component {
       ...props
     } = this.props;
```
Comment (taion) [inline src/ModalHeader.js]: we don't get an unused var thing for keeping `onHide` here?

File: src/NavItem.js [hunk 2]
```diff
@@ -28,14 +27,9 @@ class NavItem extends React.Component {
   }
 
   handleClick(e) {
-    if (this.props.disabled) {
```
Comment (taion) [inline src/NavItem.js]: wait, no more disabled handling?

File: src/PagerItem.js [hunk 3]
```diff
@@ -21,23 +20,10 @@ const defaultProps = {
...
-  constructor(props, context) {
-    super(props, context);
-
-    this.handleSelect = this.handleSelect.bind(this);
-  }
-
-  handleSelect(e) {
-    const { disabled, onSelect, eventKey } = this.props;
-
-    if (disabled) {
```
Comment (taion) [inline src/PagerItem.js]: also here

File: src/SafeAnchor.js [hunk 4]
```diff
@@ -85,7 +86,7 @@ class SafeAnchor extends React.Component {
       <Component
         {...props}
         onClick={this.handleClick}
-        onKeyDown={createChainedFunction(this.handleKeyDown, onKeyDown)}
```
Comment (taion) [inline src/SafeAnchor.js]: we can remove spreading out `onKeyDown`?

File: src/TabPane.js [hunk 5]
```diff
@@ -155,21 +154,19 @@ class TabPane extends React.Component {
     return tabContent && tabContent.animation;
   }
 
-  handleEnter() {
-    const tabContent = this.context.$bs_tabContent;
-    if (!tabContent) {
-      return;
-    }
+  handleEnter(...args) {
+    if (this.props.onEnter) this.props.onEnter(...args);
```
Comment (taion) [inline src/TabPane.js]: we should call `props.onEnter` after running our own `onEnter` handler, for consistency with elsewhere, no?

File: src/NavItem.js (hunk 2 above)
Comment (jquense) [reply src/NavItem.js]: SafeAnchor handles it already /shrug

File: src/TabPane.js (hunk 5 above)
Comment (jquense) [reply src/TabPane.js]: probably, tho this was how it was before so idk

File: src/TabPane.js [hunk 6]
```diff
@@ -155,21 +154,19 @@ class TabPane extends React.Component {
     return tabContent && tabContent.animation;
   }
 
-  handleEnter() {
-    const tabContent = this.context.$bs_tabContent;
-    if (!tabContent) {
-      return;
-    }
+  handleEnter(...args) {
```
Comment (taion) [inline src/TabPane.js]: no class properties here?

File: src/TabPane.js (hunk 5 above)
Comment (taion) [reply src/TabPane.js]: the `createChainedFunction` put `props.onEnter` after `this.handleEnter`, though.

maybe just switch up the branch below to `if (tabContent)`?

File: src/ModalHeader.js [hunk 7]
```diff
@@ -49,7 +49,6 @@ class ModalHeader extends React.Component {
     const {
       closeLabel,
       closeButton,
-      onHide,
```
Comment (taion) [inline src/ModalHeader.js]: wait that means this gets passed down to the `<div>` now?

File: src/TabPane.js [hunk 8]
```diff
@@ -154,21 +151,22 @@ class TabPane extends React.Component {
...
-    if (this.props.onExited) this.props.onExited(...args);
+    if (this.props.onEnter) this.props.onEnter(...args);
+  }
 
+  handleExited = (...args) => {
     const tabContent = this.context.$bs_tabContent;
-    if (!tabContent) return;
-    tabContent.onPaneExited(this);
-    this.in = false;
+    if (!tabContent) {
```
Comment (taion) [inline src/TabPane.js]: invert condition

**Comment by jquense** [comment]
Deploy preview ready!

Built with commit 6c925c7c3f2d9764514d90cb065e39375cbf8662

https://deploy-preview-2922--beggar-exchange-38318.netlify.com

**Comment by taion** [comment]
sorry i'm a little confused – do we not warn any more on unused things in object spreads?

**Comment by jquense** [comment]
Deploy preview ready!

Built with commit 75c8fcea1797ad1e30e89b78c7c23e6213ca37a1

https://deploy-preview-2922--beggar-exchange-38318.netlify.com

**Comment by taion** [comment]
oh gosh can we turn off the netlify comments?

**Comment by taion** [comment]
i think you need to rebase

main concern here is just making sure that we don't unexpectedly send props to the wrong places. i'm not a huge fan of having unused bindings in spread expressions, but if we're already doing this a bunch, it's not awful to keep doing so

**Comment by jquense** [comment]
Honestly I'm not  sure why they aren't lint errors, I guess that's airbnb? I did try and check that anything that was previously spread was still OK with the later handler prop after the spread (clearly I missed one tho)

**Comment by taion** [comment]
fair enough. it'd be nice to be consistent but whatever. can you merge/rebase this PR?

File: src/MenuItem.js [hunk 9]
```diff
@@ -74,7 +73,7 @@ class MenuItem extends React.Component {
   }
 
   handleClick(event) {
```
Comment (taion) [inline src/MenuItem.js]: use arrow fn?

File: package.json [hunk 10]
```diff
@@ -26,7 +26,7 @@
   },
   "lint-staged": {
     "*.js": [
-      "format",
+      "npm run ci-lint -- --fix",
```
Comment (taion) [inline package.json]: `npm run format` no good here?

File: src/NavItem.js [hunk 11]
```diff
@@ -69,11 +63,7 @@ class NavItem extends React.Component {
         className={classNames(className, { active, disabled })}
         style={style}
       >
-        <SafeAnchor
-          {...props}
-          disabled={disabled}
-          onClick={createChainedFunction(onClick, this.handleClick)}
-        />
+        <SafeAnchor {...props} disabled={disabled} onClick={this.handleClick} />
```
Comment (taion) [inline src/NavItem.js]: don't spread out `onClick` above (no need to delete since it gets overridden)

File: src/Navbar.js [hunk 12]
```diff
@@ -127,27 +126,23 @@ class Navbar extends React.Component {
...
-          collapseOnSelect ? this.handleCollapse : null
-        )
+        onSelect: this.handleCollapse
       }
     };
   }
 
-  handleCollapse() {
-    const { onToggle, expanded } = this.props;
+  handleCollapse(...args) {
```
Comment (taion) [inline src/Navbar.js]: use arrow fn properties?

File: src/NavbarToggle.js [hunk 13]
```diff
@@ -22,14 +21,20 @@ const contextTypes = {
 };
 
 class NavbarToggle extends React.Component {
+  handleClick = (...args) => {
+    const { onClick } = this.props;
+    const navbarProps = this.context.$bs_navbar;
+    if (onClick) onClick(...args);
+    if (navbarProps && navbarProps.onToggle) navbarProps.onToggle(...args);
```
Comment (taion) [inline src/NavbarToggle.js]: ooh does that fix that lingering issue with collapse breaking links?

File: src/NavbarToggle.js [hunk 14]
```diff
@@ -22,14 +21,20 @@ const contextTypes = {
...
+  };
   render() {
     const { onClick, className, children, ...props } = this.props;
```
Comment (taion) [inline src/NavbarToggle.js]: don't need to spread out `onClick` any more?

File: src/PagerItem.js [hunk 15]
```diff
@@ -62,7 +48,7 @@ class PagerItem extends React.Component {
         <SafeAnchor
           {...props}
           disabled={disabled}
-          onClick={createChainedFunction(onClick, this.handleSelect)}
+          onClick={this.handleSelect}
```
Comment (taion) [inline src/PagerItem.js]: drop `onClick` from spread?

File: src/PanelToggle.js [hunk 16]
```diff
@@ -31,26 +30,19 @@ const contextTypes = {
...
-  }
+  handleToggle = event => {
+    const { onClick } = this.props;
+    const panel = this.context.$bs_panel;
+    if (onClick) onClick(event);
+    if (panel && panel.onToggle) panel.onToggle(event);
+  };
 
   render() {
     const { onClick, className, componentClass, ...props } = this.props;
```
Comment (taion) [inline src/PanelToggle.js]: drop `onClick` from spread?

File: src/TabPane.js [hunk 17]
```diff
@@ -265,12 +260,12 @@ class TabPane extends React.Component {
       return (
         <Transition
           in={active && !exiting}
-          onEnter={createChainedFunction(this.handleEnter, onEnter)}
+          onEnter={this.handleEnter}
```
Comment (taion) [inline src/TabPane.js]: maybe add a comment in the spread indicating that those are deliberately unused, and are there to be omitted from `elementProps` above?

**Pull request closed without merging**